src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -92,113 +91,107 @@ public FilingsService(
   }
 
   public FilingApi generateOverseasEntityFiling(
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
     var filing = new FilingApi();
-    filing.setKind(FILING_KIND_OVERSEAS_ENTITY);
     setFilingApiData(filing, requestId, overseasEntityId, transaction, passThroughTokenHeader);
     return filing;
   }
 
   private void setFilingApiData(
-      FilingApi filing,
-      String requestId,
-      String overseasEntityId,
-      Transaction transaction,
-      String passThroughTokenHeader)
-      throws SubmissionNotFoundException, ServiceException {
+          FilingApi filing,
+          String requestId,
+          String overseasEntityId,
+          Transaction transaction,
+          String passThroughTokenHeader)
+          throws SubmissionNotFoundException, ServiceException {
+
     var logMap = new HashMap<String, Object>();
     logMap.put(OVERSEAS_ENTITY_ID_KEY, overseasEntityId);
     logMap.put(TRANSACTION_ID_KEY, transaction.getId());
 
     Map<String, Object> userSubmission = new HashMap<>();
 
     OverseasEntitySubmissionDto submissionDto =
-        setSubmissionData(userSubmission, overseasEntityId, logMap);
-    setPaymentData(userSubmission, transaction, passThroughTokenHeader, logMap);
+            overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId)
+                    .orElseThrow(() -> new SubmissionNotFoundException(
+                            String.format("Empty submission returned when generating filing for %s", overseasEntityId)
+                    ));
 
     if (submissionDto.isForUpdate()) {
-      publicDataRetrievalService.initialisePublicData(
-          submissionDto.getEntityNumber(), passThroughTokenHeader);
-      privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+      getPublicAndPrivateData(submissionDto.getEntityNumber(), passThroughTokenHeader);
+      //Call a method to populate UpdateSubmission
```

Comment:
Is this comment related to another ticket or can it be removed?

Reply:
For now we could instead do:
// To be replaced with UpdateSubmission model in JIRA number:
setSubmissionData(userSubmission, submissionDto, logMap); 
and re-enable the test.